src/Refitter.Core/XmlDocumentationGenerator.cs

```diff
@@ -0,0 +1,145 @@
+using System.Text;
+
+using NSwag;
+using NSwag.CodeGeneration.CSharp.Models;
+
+namespace Refitter.Core;
+
+/// <summary>
+/// Generator class for creating XML documentation.
+/// </summary>
+public class XmlDocumentationGenerator
+{
+    /// <summary>
+    /// The global code generation settings.
+    /// </summary>
+    private readonly RefitGeneratorSettings _settings;
+
+    /// <summary>
+    /// The whitespace to use for a single level of indentation.
+    /// </summary>
+    private const string Separator = "    ";
+
+    /// <summary>
+    /// Instantiates a new instance of the <see cref="XmlDocumentationGenerator"/> class.
+    /// </summary>
+    /// <param name="settings">The code generation settings to use.</param>
+    internal XmlDocumentationGenerator(RefitGeneratorSettings settings)
+    {
+        this._settings = settings;
+    }
+
+    /// <summary>
+    /// Appends XML docs for the given interface definition to the given code builder.
+    /// </summary>
+    /// <param name="group">The OpenAPI definition of the interface.</param>
+    /// <param name="code">The builder to append the documentation to.</param>
+    public void AppendInterfaceDocumentation(OpenApiOperation group, StringBuilder code)
+    {
+        if (!_settings.GenerateXmlDocCodeComments || string.IsNullOrWhiteSpace(group.Summary))
+            return;
+
+        this.AppendXmlCommentBlock("summary", group.Summary, code, indent: Separator);
+    }
+
+    /// <summary>
+    /// Appends XML docs for the given method to the given code builder.
+    /// </summary>
+    /// <param name="method">The NSwag model of the method's OpenAPI definition.</param>
+    /// <param name="code">The builder to append the documentation to.</param>
+    public void AppendMethodDocumentation(CSharpOperationModel method, StringBuilder code)
+    {
+        if (!_settings.GenerateXmlDocCodeComments)
+            return;
+
+        if (!string.IsNullOrWhiteSpace(method.Summary))
+            this.AppendXmlCommentBlock("summary", method.Summary, code);
+
+        if (!string.IsNullOrWhiteSpace(method.Description))
+        {
+            this.AppendXmlCommentBlock("remarks", method.Description, code);
+        }
+
+        foreach (var parameter in method.Parameters)
+        {
+            if (parameter == null || string.IsNullOrWhiteSpace(parameter.Description))
+                continue;
+
+            this.AppendXmlCommentBlock("param", parameter.Description, code, new Dictionary<string, string>
+                { ["name"] = parameter.VariableName });
+        }
+
+        if (method.HasResult && !string.IsNullOrWhiteSpace(method.ResultDescription))
```

Comment:
Changing

```csharp
        if (method.HasResult && !string.IsNullOrWhiteSpace(method.ResultDescription))
        {
            this.AppendXmlCommentBlock("returns", method.ResultDescription, code);
        }

        this.AppendXmlCommentBlock("throws", this.BuildErrorDescription(method.Responses), code, new Dictionary<string, string>
            { ["cref"] = "ApiException" });
```

into

```csharp
        if (_settings.ReturnIApiResponse)
        {
            this.AppendXmlCommentBlock("returns", "An instance of IApiResponse", code);
        }
        else
        {
            if (method.HasResult && !string.IsNullOrWhiteSpace(method.ResultDescription))
            {
                this.AppendXmlCommentBlock("returns", method.ResultDescription, code);
            }
        }

        if (!_settings.ReturnIApiResponse)
            this.AppendXmlCommentBlock("throws", this.BuildErrorDescription(method.Responses), code, new Dictionary<string, string>
                { ["cref"] = "ApiException" });
```

will generate the `<throws />` XML docs when the success response type is directly returned

```csharp
        /// <summary>Add a new pet to the store</summary>
        /// <remarks>Add a new pet to the store</remarks>
        /// <param name="body">Create a new pet in the store</param>
        /// <returns>Successful operation</returns>
        /// <throws cref="ApiException">
        /// Thrown when the request returns a non-success status code:
        /// 405: Invalid input
        /// </throws>
        [Headers("Accept: application/xml, application/json")]
        [Post("/pet")]
        Task<Pet> AddPet([Body] Pet body);
```

and when the return type is `IApiResponse` or `IApiResponse<T>` then the generated comments will look like this:

```csharp
        /// <summary>Add a new pet to the store</summary>
        /// <remarks>Add a new pet to the store</remarks>
        /// <param name="body">Create a new pet in the store</param>
        /// <returns>An instance of IApiResponse</returns>
        [Headers("Accept: application/xml, application/json")]
        [Post("/pet")]
        Task<IApiResponse<Pet>> AddPet([Body] Pet body);
```

Reply:
That would be great @DJ4ddi